presto-main/src/main/java/com/facebook/presto/execution/SetSessionTask.java

```diff
@@ -54,12 +57,18 @@ else if (propertyName.getParts().size() == 2) {
 
         Type type = propertyMetadata.getSqlType();
 
-        Object objectValue = evaluatePropertyValue(statement.getValue(), type, session, metadata);
-        String value = serializeSessionProperty(type, objectValue);
+        try {
+            Object objectValue = evaluatePropertyValue(statement.getValue(), type, session, metadata);
+            String value = serializeSessionProperty(type, objectValue);
 
-        // verify the SQL value can be decoded by the property
-        metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
+            // verify the SQL value can be decoded by the property
+            metadata.getSessionPropertyManager().decodeProperty(propertyName.toString(), value, propertyMetadata.getJavaType());
 
-        stateMachine.addSetSessionProperties(propertyName.toString(), value);
+            stateMachine.addSetSessionProperties(propertyName.toString(), value);
+        }
+        catch (SemanticException e) {
```

Comment:
This `catch` is too broad. Which of the statements above can throw it?

Also:
* `SessionPropertyManager.decodeProperty` already validates the name, so this seems like an unrelated change
* The commit is supposed to be about validating table properties. This method handles setting session properties.